src/backend/app/Http/Controllers/VodacomTransactionController.php

```diff
@@ -2,10 +2,18 @@
 
 namespace App\Http\Controllers;
 
+use App\Http\Resources\VodacomResource;
 use App\Models\Transaction\VodacomTransaction;
+use App\Services\VodacomService;
 use Illuminate\Http\Request;
 
 class VodacomTransactionController extends Controller {
+    private VodacomService $vodacomService;
+
+    public function __construct(VodacomService $vodacomService) {
+        $this->vodacomService = $vodacomService;
+    }
```

Comment:
You can use (so-called) constructor property promotion here to simplify this to

```suggestion
    public function __construct(
        private VodacomService $vodacomService
    ) {}
```

Reply:
Updated.